daemon/containerd/image_pull.go

```diff
@@ -217,6 +217,18 @@ func (i *ImageService) pullTag(ctx context.Context, ref reference.Named, platfor
 			}
 			return errdefs.NotFound(fmt.Errorf("pull access denied for %s, repository does not exist or may require 'docker login'", reference.FamiliarName(ref)))
 		}
+		if cerrdefs.IsNotFound(err) {
+			// Transform "no match for platform in manifest" error returned by containerd into
+			// the same message as the graphdrivers backend.
+			// The one returned by containerd doesn't contain the platform and is much less informative.
+			if strings.Contains(err.Error(), "platform") {
+				platformStr := platforms.DefaultString()
```

Comment:
Just a note; this also includes the os-version on Windows. I don't think we show that in other places, but perhaps it's good to show, as we would be matching using that (I think?)

Reply:
Yes, IMO it makes sense to show it. Although it will only really be matched on the windows host due to different containerd platform matcher implementation based on the target platform the code was built for.